integration/simple-service/simple-service-test.ts

```diff
@@ -0,0 +1,69 @@
+import { SimpleServiceClientImpl, SimpleServiceJsonClientImpl } from './simple_service';
+import { Simple, StateEnum, Child_Type } from './simple';
+
+const simple: Simple = {
+  name: 'asdf',
+  age: 1,
+  child: { name: 'child', type: Child_Type.UNKNOWN },
+  state: StateEnum.ON,
+  grandChildren: [
+    { name: 'grand1', type: Child_Type.UNKNOWN },
+    { name: 'grand2', type: Child_Type.UNKNOWN },
+  ],
+  coins: [2, 4, 6],
+  snacks: ['a', 'b'],
+  oldStates: [StateEnum.ON, StateEnum.OFF],
+  createdAt: new Date('1970-01-01T00:00:00.000Z'),
+  thing: undefined,
+  blobs: [],
+};
+
+describe('simple service', () => {
+  it('can echo with protobuf serialization', async () => {
+    const echoRpc = {
+      request: (_service: string, _method: string, data: Uint8Array) => {
```

Comment:
Ah sure, that makes sense. How about using `jest.fn`s though so we can assert the right values are passed? I.e. `expect(...).toBeCalledWith`?